### DISCUSSION TITLE
esportare  un modello dati da documento esistente  ( bottone getTheCall  )

### DISCUSSION THREAD

**Discussion by lhenry2k** [fattureincloud/api#454] [Q&A] [answered]
Buongiorno, 
ho letto lo schema "create issued doc", ma vedo l'operazione di scelta e fill dei parametri immensa rispetto allo "sforzo" per la chiamata.

Nel mio caso devo inviare mensilmente diverse TD17 uguali ogni mese ma sarebbe applicabile a qualunque altra operazione.

Sarebbe fantastico avere un nuovo bottone "show Json calls" a fianco a "modifica, duplica, apri " , un po come tutti le piu recenti API playground  consoles, ma partendo da un documento gia emesso con la comoditá della UI.

A questo punto basterebbe cambiare i pochi dati variabili e ciclare le curl , previa creazione del Bearer

**Comment by lhenry2k** [comment]
Sul vs sito  leggo questa frase in importa/esporta > Fattura  (xmls zippati)

_Le fatture generate a partire da questi documenti non potranno essere inviate al Sistema di Interscambio_

Se invece fosse possibile, questa sarebbe una molto facile , veloce e alla portata di molti 

grazie

**Comment by valmoz** [comment, marked as answer]
Buongiorno @lhenry2k, le rispondo per punti.

1. Se ho capito bene la sua richiesta iniziale lei vorrebbe ottenere il JSON di creazione del documento a partire da un documento già esistente. In tal caso, il nostro suggerimento è quello di effettuare una chiamata GET sul documento che si intende copiare, e usare il JSON ottenuto come base per la creazione di nuovi documenti. Di fatto gli schema del payload della Create e del body della Get sono identici, fatto salvo alcuni campi (ad esempio l'ID del documento) che vengono impostati dal nostro software in fase di creazione e non possono essere conosciuti a priori.
2. Per quanto riguarda la funzionalità di invio all'SDI di documenti importati tramite XML, la avviso che questa funzionalità non verrà implementata come anche specificato su [questa guida](https://help.fattureincloud.it/help/articolo/541-importa-fatture-anagrafiche-altri-documenti-altri-software); la funzionalità di import tramite XML è stata infatti pensata esclusivamente come import dello storico delle fatture create utilizzando software terzi. L'invio di fatture all'SDI rimarrà utilizzabile solo per le fatture create direttamente tramite Fatture in Cloud. Nulla però le vieta di creare un'applicazione custom che si occupi di leggere gli XML delle fatture e di importarle in Fatture in Cloud tramite le nostre API; in tal caso le fatture risulterebbero tranquillamente inviabili all'SDI.

A disposizione per eventuali chiarimenti
Mauro

**Reply by lhenry2k** [reply]
la 1) risolve la 2) , ottimo,  prenderó questa strada.
grazie !!

**Reply by valmoz** [reply]
Lieto di esserle stato utile!

Grazie
Mauro

**Comment by lhenry2k** [comment]
Recappo qui i campi che penso vadano esclusi dal Json di lettura  per creare il Json di scrittura in quanto generati al momento dell'invio 

- data.id
- data.items_list.id
- data.payments_list.id
- data.created_at
- data.updated_at

usando invece un fornitore giá noto as Fic , credo che 

- data.entity.id

possa essere specificato, assieme a tutti i campi descrittivi del fornitore in quanto leggo che FiC non fa (ancora) un lookup

**Reply by valmoz** [reply]
Buongiorno @lhenry2k 
per quanto riguarda l'ID della Entity, la logica è la seguente:

- Se il cliente/fornitore è già censito nell'anagrafica di Fatture in Cloud, allora inserendo l'ID sarà possibile creare il collegamento tra il cliente/fornitore e il documento
- Se il cliente non è censito (esempio cliente non ricorrente), basterà ignorare l'ID per procedere comunque.

In entrambi i casi, è sempre necessario inserire gli altri dettagli in quanto non recuperiamo le informazioni a partire dall'ID.

Può leggere ulteriori informazioni qui: [Invoice Creation](https://developers.fattureincloud.it/docs/guides/invoice-creation#one-step-one-the-customer)
Consiglio anche di leggere la nota relativa all'indirizzo email presente nelle [FAQ](https://developers.fattureincloud.it/docs/FAQs#how-can-i-show-the-client's-mail,-fax,-etc.-in-the-document).

Grazie
Mauro
